cmd/abapEnvironmentPushATCSystemConfig.go

```diff
@@ -0,0 +1,235 @@
+	// Unit tests shall be executable in parallel (not depend on global state), and don't (re-)test dependencies.
+}
+
+type abapEnvironmentPushATCSystemConfigUtilsBundle struct {
+	*command.Command
+	*piperutils.Files
+
+	// Embed more structs as necessary to implement methods or interfaces you add to abapEnvironmentPushATCSystemConfigUtils.
+	// Structs embedded in this way must each have a unique set of methods attached.
+	// If there is no struct which implements the method you need, attach the method to
+	// abapEnvironmentPushATCSystemConfigUtilsBundle and forward to the implementation of the dependency.
+}
+
+func newAbapEnvironmentPushATCSystemConfigUtils() abapEnvironmentPushATCSystemConfigUtils {
+	utils := abapEnvironmentPushATCSystemConfigUtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+
+	return &utils
+}
+
+func abapEnvironmentPushATCSystemConfig(config abapEnvironmentPushATCSystemConfigOptions, telemetryData *telemetry.CustomData) {
+	// for command execution use Command
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+
+	var autils = abaputils.AbapUtils{
+		Exec: &c,
+	}
+
+	client := piperhttp.Client{}
+
+	err := runAbapEnvironmentPushATCSystemConfig(&config, telemetryData, &autils, &client)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAbapEnvironmentPushATCSystemConfig(config *abapEnvironmentPushATCSystemConfigOptions, telemetryData *telemetry.CustomData, autils abaputils.Communication, client piperhttp.Sender) error {
+
+	subOptions := convertATCSysOptions(config)
+
+	// Determine the host, user and password, either via the input parameters or via a cloud foundry service key
+	connectionDetails, err := autils.GetAbapCommunicationArrangementInfo(subOptions, "/sap/opu/odata4/sap/satc_ci_cf_api/srvd_a2x/sap/satc_ci_cf_sv_api/0001")
+	if err != nil {
+		return errors.Wrap(err, "Parameters for the ABAP Connection not available")
+	}
+
+	cookieJar, err := cookiejar.New(nil)
+	if err != nil {
+		return errors.Wrap(err, "Could not create a Cookie Jar")
+	}
+	clientOptions := piperhttp.ClientOptions{
+		MaxRequestDuration: 180 * time.Second,
+		CookieJar:          cookieJar,
+		Username:           connectionDetails.User,
+		Password:           connectionDetails.Password,
+	}
+	client.SetOptions(clientOptions)
+
+	return pushATCSystemConfig(config, connectionDetails, client)
+
+}
+
+func pushATCSystemConfig(config *abapEnvironmentPushATCSystemConfigOptions, connectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender) error {
+
+	filelocation, err := filepath.Glob(config.AtcSystemConfigFilePath)
+	//check ATC system configuration json
+	var atcSystemConfiguartionJsonFile []byte
+	if err == nil {
```

Comment:
My (personal) preference is to return early and to avoid "if err == nil" cascades (even though you may find it in some steps). But if you prefer this way, I'm ok with it.